lib/sneakers/worker.rb

```diff
@@ -52,10 +53,11 @@ def do_work(delivery_info, metadata, msg, handler)
           metrics.increment("work.#{self.class.name}.started")
           Timeout.timeout(@timeout_after, Timeout::Error) do
             metrics.timing("work.#{self.class.name}.time") do
+              deserialized_msg = ContentType[@content_type || metadata[:content_type]].deserialize(msg)
```

Comment:
There's no docs on this method indicating metadata will always be present and it's not private so you might need to handle the `metadata == nil` case. 

Separately I think there are cases where you would want to prefer the worker's content type over the message's and cases for the inverse. Might be worth refactoring that bit of logic into a method that respects a new preference set on the worker indicating which order to prefer.